pkg/nsx/services/inventory/pod.go

```diff
@@ -0,0 +1,84 @@
+package inventory
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/antihax/optional"
+	nsxt "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/containerinventory"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (s *InventoryService) SyncContainerApplicationInstance(name string, namespace string, key InventoryKey) *InventoryKey {
+	pod := &corev1.Pod{}
+	err := s.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, pod)
+	externalId := key.ExternalId
+	if apierrors.IsNotFound(err) ||
+		((err == nil) && (string(pod.UID) != externalId)) {
+		err = s.DeleteResource(externalId, ContainerApplicationInstance)
+		if err != nil {
+			log.Error(err, "Delete ContainerApplicationInstance Resource error", "key", key)
+			return &key
+		}
+	} else if err == nil {
+		retry := s.BuildPod(pod)
+		if retry {
+			return &key
+		}
+	} else {
+		log.Error(err, "Unexpected error is found while processing Pod")
+	}
+	return nil
+}
+
+func (s *InventoryService) syncContainerApplicationInstance(clusterId string) error {
```

Comment:
Is this function for initializing the ContainerApplicationInstance store? Maybe use a different name to distinguish from SyncContainerApplicationInstance above, like initContainerApplicationInstanceStore?

Reply:
agree.  it's sync from NSX, the other is sync to NSX. using initContainerApplicationInstanceStore